scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala

```diff
@@ -374,10 +362,28 @@ trait PrinterOps { self: DatabaseOps =>
             this.printType(s.typeSignature)
             this.print(s.flagsExplanationString)
           }
+          if (!result.isInstanceOf[MethodType]) {
+            // Only print `: ` before last curried parameter list.
+            this.print(": ")
+          }
           this.printType(result)
+        case NullaryMethodType(result) =>
+          printType(result)
```

Comment:
How about `: $result`?

Reply:
Good idea, but it seems to trigger for `val`s too. I propose we keep it unchanged since this is enough to get ExplicitReturnTypes working for defs, but consider adding this later for `def`s.

Reply:
Should I open an issue for this? I agree the denotation.signature for nullary methods should be `Type.Method(Nil, _)`